core/src/main/java/org/apache/iceberg/TableMetadataParser.java

```diff
@@ -262,7 +277,41 @@ static TableMetadata fromJson(FileIO io, InputFile file, JsonNode node) {
       lastSequenceNumber = TableMetadata.INITIAL_SEQUENCE_NUMBER;
     }
     int lastAssignedColumnId = JsonUtil.getInt(LAST_COLUMN_ID, node);
-    Schema schema = SchemaParser.fromJson(node.get(SCHEMA));
+
+    List<Schema> schemas;
+    int currentSchemaId;
+    Schema schema = null;
```

Comment:
Why set this to null but not the two others?

Reply:
This is because in the first if statement, after parsing the schema array and assign the schema, I did a check to ensure the schema holding current schema ID exists in schema array. Without this null check, the `Preconditions.checkArgument(schema != null, ...)` will highlight with error warning "schema variable might not be initialized".